### ISSUE TITLE
Added Ragnarok instance to clusterd setup Aborted Bad PID

### ISSUE THREAD

**Issue by linkuso** [arkmanager/ark-server-tools#1277]
Hi i'm trying to add Ragnarok to a excisting clusterd setup that a friend of mine (that unfortunetly has passed away) have setup, this is my first time using tinkering with arkmanager and clusterd setups.

Some info
i have run 

**arkmanager installmods --verbose @ragnarok**

i have added Ragnarok to a new line in arkmanager.cfg
configfile_ragnarok="/etc/arkmanager/instances/ragnarok.cfg"

**the ragnarok.cfg looks like this**
`arkserverroot="/home/steam/Servers/Ragnarok"
serverMap="Ragnarok"
ark_Port=7791
ark_QueryPort=27029
ark_RCONPort=27030
ark_AltSaveDirectoryName="Ragnarok"
ark_ServerPassword=""
`

**arkserverroot folder looks like this**
`/home/steam/Servers/Ragnarok
steam@ark:~/Servers/Ragnarok$ ls -la
total 85412
drwxr-xr-x  6 steam steam     4096 Jul 12 12:30 .
drwxr-xr-x 11 steam steam     4096 Jul 12 12:23 ..
drwxr-xr-x  7 steam steam     4096 Jun 11  2020 Engine
drwxr-xr-x  2 steam steam     4096 Jul 12 11:01 linux64
-rwxr-xr-x  7 steam steam     1987 Jul 30  2022 Manifest_DebugFiles.txt
-rwxr-xr-x  7 steam steam    19609 Jul 30  2022 Manifest_NonUFSFiles.txt
-rwxr-xr-x  7 steam steam    73578 Jul 30  2022 Manifest_UFSFiles.txt
-rwxr-xr-x  3 steam steam  5610607 Mar 24 03:30 PackageInfo.bin
drwxr-xr-x  6 steam steam     4096 Sep  1  2021 ShooterGame
-rw-r--r--  1 steam steam        0 Jul 12 12:39 .startAfterUpdate-ragnarok
drwxr-xr-x  5 steam steam     4096 Jul 12 11:01 steamapps
-rwxr-xr-x  7 steam steam 21576016 Dec 27  2021 steamclient64.dll
-rwxr-xr-x  7 steam steam 18262952 Dec 27  2021 steamclient.dll
-rwxr-xr-x  2 steam steam 40226316 Jul 12 11:01 steamclient.so
-rwxr-xr-x  7 steam steam      272 Aug  5  2021 SteamCMDInstall.sh
-rwxr-xr-x  7 steam steam   403368 Dec 27  2021 tier0_s64.dll
-rwxr-xr-x  7 steam steam   350120 Dec 27  2021 tier0_s.dll
-rwxr-xr-x  3 steam steam        9 Mar 24 03:30 version.txt
-rwxr-xr-x  7 steam steam   503720 Dec 27  2021 vstdlib_s64.dll
-rwxr-xr-x  7 steam steam   389032 Dec 27  2021 vstdlib_s.dll
steam@ark:~/Servers/Ragnarok$`

**Startup for Rag looks like this**
`steam@ark:~$ arkmanager start --verbose --no-background @ragnarok
Running command 'start' for instance 'ragnarok'
The server is starting...

2024-07-12 12:39:57: start
2024-07-12 12:39:57: Running /home/steam/Servers/Ragnarok/ShooterGame/Binaries/Linux/ShooterGameServer Ragnarok\?Port=7790\?QueryPort=27029\?RCONPort=27030\?AltSaveDirectoryName=Ragnarok\?ServerPassword\?AllowFlyingStaminaRecovery=True\?AllowHitMarkers=True\?AllowRaidDinoFeeding=True\?AllowThirdPersonPlayer=True\?AlwaysNotifyPlayerLeft=True\?DisableWeatherFog=True\?DontAlwaysNotifyPlayerJoined=False\?GameModIds=893404454\,1999447172\,895711211\,640022028\,1404697612\,821530042\,893834064\,1148802682\,845781509\,924619115\,1814953878\,2044129379\,1609138312\?GlobalVoiceChat=False\?MaxPlayers=20\?NoTributeDownloads=False\?OverrideStructurePlatformPrevention=True\?PreventSpawnAnimations=True\?RCONEnabled=True\?ServerAdminPassword=<omited>\?ServerForceNoHUD=False\?ServerHardcore=False\?ServerPVE=False\?ServerPassword\?ShowFloatingDamageText=True\?ShowMapPlayerLocation=True\?UseOptimizedHarvestingHealth=True\?bAllowFlyerCarryPvE=True\?bDisableStructureDecayPvE=True\?bDisableStructurePlacementCollision=True\?bUseCorpseLocator=True\?listen -Log -NoTransferFromFiltering -Server -ServerRCONOutputTribeLogs -USEALLAVAILABLECORES -servergamelog -servergamelogincludetribelogs -clusterDirOverride=/home/steam/Steam/arkcluster/ -clusterId=ofdeathark
2024-07-12 12:39:57: Server PID: 816087
Shutdown handler: initalize.
Using binned.
4.5.1-0+UE4 7038 3077 404 10
[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
/usr/local/bin/arkmanager: line 1314: 816087 Aborted                 "$arkserverroot/$arkserverexec" "$arkserveropts" "${arkextraopts[@]}"
2024-07-12 12:40:02: Bad PID ''; expected '816087'
2024-07-12 12:40:02: exited with status 0
steam@ark:~$`

Do i need to add the Instance to the cluster or something else that i have missed? The logs aret really telling me much? Can i look somewhere elese?

Thanks

**Comment by klightspeed** [comment]
Condolences on your loss.

Did the server log anything in `/home/steam/Servers/Ragnarok/ShooterGame/Saved/Logs/ShooterGame.log`?

The log suggests that the game server is exiting within 5 seconds of starting.

Using `lsof /home/steam/Servers/Ragnarok`, are there any processes running in or keeping files open in the game server directory?

The ARK server does sometimes accidentally daemonize itself (by re-executing itself and exiting), especially when it has automanagedmods enabled. There was a fix to restart the server when this happens, which I had forgotten to create a release for. This should now be in v1.6.63

If there is still a `ShooterGameServer` process running, you may need to `kill` it.

**Comment by linkuso** [comment]
Thankyou,

The server hasent logged anything in the Saved folder for Rag, hasn't even created the Logs folder just Config folder and an empty Ragnarok folder. :/

**Comment by klightspeed** [comment]
Given it sounds like this instance is being started for the first time, with no existing save, does it work without mods (or at least with the same mod set as is on your TheIsland instance)?

**Comment by linkuso** [comment]
Yes that i would assume aswell, the folderstructure did arkmanager create when i ran 
**arkmanager installmods --verbose @ragnarok**

If im comparing the config files to TheIsland i cannot see anywhere other than in arkmanager.cfg and a mods.txt file (that are in the same folder as arkmanager.cfg where) Mod's ar defined, i dont know if the mods.txt is something that arkmanager uses or if that is something that my friend made to just note for himself what mods ar running.

**arkmanager.cfg**

> ark_GameModIds="893404454,1999447172,895711211,640022028,1404697612,821530042,893834064,1148802682,845781509,924619115,1814953878,2044129379,1609138312"

I just realized somehing... do i have to define Ragnarok in ark_GameModIds ?

**Comment by klightspeed** [comment]
> If im comparing the config files to TheIsland i cannot see anywhere other than in arkmanager.cfg and a mods.txt file (that are in the same folder as arkmanager.cfg where) Mod's ar defined, i dont know if the mods.txt is something that arkmanager uses or if that is something that my friend made to just note for himself what mods ar running.

Unless it is read in by one of the config files, `mods.txt` will have no effect on arkmanager.

> I just realized somehing... do i have to define Ragnarok in ark_GameModIds ?

ARK DLC maps do not need to be referenced in `MapModId` or `GameModIds`

> And no i dont think it has any save, do i need to transfer a excisting save over?

No, provided the server files are properly up-to-date (which you can ensure by running `arkmanager install --verbose @ragnarok`, which runs `steamcmd ... app_update 376030 validate`), the game server should create the world save file on the first SaveWorld or graceful exit, and should create config files such as `ShooterGame/Saved/Config/LinuxServer/GameUserSettings.ini`

**Comment by linkuso** [comment]
OH that reminds me, the GameUserSetting.ini and Game.ini where empty before and i copied TheIslands files, i'll try to remove them and rerun the verbose installer.

Btw is it possible to get some kind of verbose output from steamcmd when staerting the server in no-background mode?

**Comment by linkuso** [comment]
Holy shit i actually think arkmanager install --verbose @ragnarok did it!

> steam@ark:~$ arkmanager start --no-background @ragnarok
> Running command 'start' for instance 'ragnarok'
> Start aborted due to server already running - pid: 2076836


> steam@ark:~$ arkmanager status @ragnarok
> Running command 'status' for instance 'ragnarok'
>  Server running: Yes
>  Server PID: 2076836
>  Server listening: Yes
> Server Name: ARK #27270 - (v358.24)
> Steam Players: 0 / 20
> Active Steam Players: 0
>  Server online: Yes
>  ARKServers link: http://arkservers.net/server/myip:andport
>  Steam connect link: steam://connect/my-ip
>  Server build ID: 13834083
> Server version: 358.24
> steam@ark:~$

**Comment by klightspeed** [comment]
> Btw is it possible to get some kind of verbose output from steamcmd when staerting the server in no-background mode?

`arkmanager run @ragnarok` should do that.
